pkg/alertmanager/state_replication_test.go

```diff
@@ -64,7 +64,8 @@ func TestStateReplication(t *testing.T) {
 				return 0
 			}
 
-			s := newReplicatedStates("user-1", tt.replicationFactor, replicationFunc, positionFunc, make(chan struct{}), log.NewNopLogger(), reg)
+			s := newReplicatedStates("user-1", tt.replicationFactor, replicationFunc, positionFunc, log.NewNopLogger(), reg)
+			require.NoError(t, s.Service.StartAsync(context.Background()))
```

Comment:
I would also wait until started:

```suggestion
			require.NoError(t, services.StartAndAwaitRunning(ctx, s))
```